pkgs/fc/agent/fc/util/enc.py

```diff
@@ -222,25 +222,38 @@ def get_release_info(log, enc):
     release_name = params.get("release_name")
     known_releases = [r["release_name"] for r in releases.values()]
 
+    # We treat named releases as immutable, which means that we won't update
+    # release metadata objects once they are written to `release_info_path`.
+    # It's possible to change metadata for a build version by adding a new
```

Comment:
After going through the whole enc retrieval flow I can confirm that updating the release metadata will replace the existing data of a release with the same version.
Good.